PR을 위한 기존 코드 Commit

log1995 opened the pull request.

log1995 committed:
PR을 위한 기존 코드 Commit

lleellee0 commented on src/main/java/com/log1995/javacalculator/controller/CalculatorController.java:
```diff
@@ -0,0 +1,41 @@
+                    String result = calculatorService.calculateFormula(formula);
+                    calculatorConsole.printResult(result);
+                break;
+            }
```
사용자가 num으로 3을 입력했을 때는 어떻게되어야할까?
물론 요구사항에는 처리할 내용이 나와있지 않긴 하지만 어떻게 나오면 좋을 것 같음

log1995 replied:
"잘못 입력하였습니다" 혹은 종료 시켜주면 될 것 같은데, 3을 입력했을 때는 종료, 그 외의 것들에는 "잘못 입력하였습니다"가 출력되게 수정해볼께

lleellee0 commented on src/main/java/com/log1995/javacalculator/io/Input.java:
```diff
@@ -0,0 +1,7 @@
+public interface Input {
+
+    String inputNum();
+    String inputFormula();
```
Input, Output은 인터페이스를 선언하고, 다른 클래스들은 인터페이스가 없이 바로 구현체를 만들어줬는데 무슨 기준인지?

lleellee0 commented on src/main/java/com/log1995/javacalculator/service/CalculatorService.java:
```diff
@@ -0,0 +1,61 @@
+
+                if(oper.equals("*")){
+                    list.add(i - 1, String.valueOf(first * second));
+                }else if(oper.equals("/")){
```
이런 if, else if문을 없앨 수 있는 방법은 없을까?

log1995 replied:
각각의 계산에 대해서 따로 메서드를 만들어서 없앨 수 있을 거 같아

lleellee0 commented on src/main/java/com/log1995/javacalculator/service/CalculatorService.java:
```diff
@@ -0,0 +1,61 @@
+
+    public String calculateFormula(String formula){
+
+        String[] formulaArr = formula.split("\\s+");
```
사용자가 수식(formula)을 잘못 입력하는 경우에 대한 체크는 필요 없을지?

log1995 replied:
이건 구현해보겠음

lleellee0 commented on src/main/java/com/log1995/javacalculator/controller/CalculatorController.java:
```diff
@@ -0,0 +1,41 @@
+                    resultData = calculatorService.getData();
+                    calculatorConsole.printRepository(resultData);
+                break;
+                case "2":
```
num이 "1", "2" 가 무슨 의미인지 코드만 보고 알 수 있을까?
1, 2가 어떤 의미인지 더 명확하게 해줄 수 있는 방법이 있을지?

log1995 replied:
final 키워드를 사용하거나 enum 클래스를 만들어서 바꿔주면 될거 같아

log1995 committed:
첫 번째 리뷰 코드 수정
  - num이 3, 혹은 다른 것이 들어왔을 때 처리해주는 코드 구현

log1995 committed:
세 번째 리뷰 코드 수정
  - 계산을 위한 enum 클래스 생성
  - 각 계산에 우선도(priority)를 부여해서 곱하기/나누기를 우선적으로 처리
  - 기존 ArrayList 대신 Stack읉 통해 수식 처리

log1995 committed:
네 번째, 다섯 번째 리뷰 코드 수정
  - 정규식을 통한 수식 확인을 위한 checkFormula 메서드 생성
  - 잘못된 수식 입력 시 자체 예외인 InvalidFormulaException 예외 발생

  - CalculatorController의 switch구문에 숫자 대신 상수를 넣어 각 번호의 의미를 명확하게 만들어줌
